app/models/concerns/digital_object_concerns/create_and_update_terms.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module DigitalObjectConcerns
+  module CreateAndUpdateTerms
+    extend ActiveSupport::Concern
+
+    private
+
+      # For all the terms in the :rights and :descriptive_metadata fields, creates the terms if they
+      # don't already exist. If the terms already exists updates the term with any new field data. Only
+      # term fields that don't have data will be updated. Errors will be raised instead of placed in the
+      # errors array.
+      def create_and_update_terms
+        # Extract for rights and descriptive_metadata. If digital object is not of the appropriate type
+        # do not extract rights terms.
+        rights_terms = if self.is_a?(DigitalObject::Item) || self.is_a?(DigitalObject::Asset)
```

Comment:
That's probably a good idea. Maybe a method on each of the classes that's something like `can_have_rights?`. Idk that's a terrible name, but I want to differentiate between "is the rights field populated" and "can the rights field be populated at all." I can ticket this for later if @elohanlon is in favor of something like this.

Reply:
@cgalarza One other option would be to defer the decision and only add `before_save :create_and_update_terms` on the Item and Asset classes, and have this method raise an error if called on another class type.  Might be the easiest way to go for now because the future of rights stuff for other objects is unknown?

Reply:
Though I'm also fine with keeping the code as is, or going with your method suggestion.  It just depends on how much future planning we want to do around currently-fuzzy future use cases.

Reply:
Here's one other suggestion: You could hypothetically change the method signature of `Hyacinth::DynamicFieldsMap.new("#{digital_object_type}_rights")` to take a second parameter that is the digital object type, and then all of the type-related stuff would be handled internally.  Then you could get rid of the `if` statement here and also keep the `before_save :create_and_update_terms` on the Digital Object base class.  I think I like that best, but totally up to you.

Reply:
`:create_and_update_terms` updates both descriptive and rights metadata, so I would have to separate this method to implement your suggestion. And that change would force us to make two solr queries for terms. Additionally, I think Ben was also referring to the other places where we check to see if a DigitalObject can have rights. For example, when validating descriptive and rights metadata.

Reply:
Oh, good point about descriptive too.  Never mind about moving the before_save callback away then.  How do you feel about my other suggestion?  Just posted it, though I think our comments may have overlapped.

Reply:
Re changing DynamicFieldsMap signature: I can't actually do that because in some cases I retrieve want to retrieve the map for all the rights fields (item and asset rights) and sometimes only item rights fields. For example, in the typedef, i have to retrieve both item and asset rights fields because I don't know the digital object type at that point.

Reply:
In the above case, are you only retrieving the type-specific rights because it's a little more efficient?  Could you just always try to retrieve all rights fields all the time here?  (if the efficiency tradeoff isn't a big deal)

Reply:
When I need to create a map for item_rights fields, its not about efficiency its actually about validating the fields that should be present in the json structure so I can't have both asset and item fields.

Reply:
Oh okay.  In that case, I'm good with either the current code or your method suggestion.

Reply:
I think I will attempt and add the `can_have_rights?` method. In order to help simplify the logic in other places.